### DISCUSSION TITLE
LOW: Workflow to add PR to merge queue is slightly odd for Github Apps

### DISCUSSION THREAD

**Discussion by purajit** [community/community#62599] [Pull Requests] [answered]
With merge queue enabled, any used can just use `gh pr merge` at any point to queue up their PR when it's ready. That's fantastic! However, using Github App credentials, the behavior is quite different:

```sh
# auth'd as myself
$ gh pr merge
✓ Pull request #17803 will be added to the merge queue for main when ready

# re-auth using jwt as a Github App
$ gh auth status
github.com
  ✓ Logged in to github.com as <...>[bot]

$ gh pr merge
X Pull request #17804 is not mergeable: the base branch policy prohibits the merge.

$ gh pr merge --auto -s
! The merge strategy for main is set by the merge queue
✓ Pull request #17804 will be added to the merge queue for main when ready
```

Even though squash & merge is the only method we allow. 

So ultimately, Github Apps need to do `gh pr merge --auto -s # (or whichever method)`, instead of `gh pr merge` like everyone else, and the CLI clearly has some underlying understanding that the merge strategy should not be passed in.

It seems like the GraphQL API used in https://github.com/cli/cli/blob/v2.32.0/internal/featuredetection/feature_detection.go#L101 isn't returning the field `isInMergeQueue` for the Github App
```
{
  __type(name: "PullRequest") {
    fields(includeDeprecated: true) {
      name
    }
  }
}
```

**Comment by ZeeshanMukhtar1** [comment]
Got it @purajit !

GitHub Apps should use `gh pr merge --auto -s` to queue up pull requests for merge with the allowed "Squash and merge" method. This avoids unnecessary prompts and streamlines the process.

**Reply by purajit** [reply]
Considering that `gh pr merge` works noninteractively without prompts when merge queue is enabled, why is the behavior different for Apps?

See the updated shell logs in my post.

**Comment by purajit** [comment, marked as answer]
I've root-caused it (at least, as much as possible client-side) and added an update on cli/cli#7213.

Relevant discussions are in
* cli/cli#7143
* cli/cli#7213

I'm going to close this since it's being tracked, even though `cli/cli` isn't the perfect place for this since it's not an issue with the cli, but rather the GraphQL server.
